### PR TITLE
fix(otel): Missing `addLink` method and Fiber handling

### PR DESCRIPTION
**Pull request by PROFeNoM**
### Description

- `SpanInterface::addLink` was added with https://github.com/open-telemetry/opentelemetry-php/pull/1289 and is included in 1.1+. It is currently missing from our shim. For backward compatibility reasons, `ImmutableSpan` will take a new argument, `totalRecordedLinks` if the latter method exists.
- https://github.com/open-telemetry/opentelemetry-php/pull/1270 modified Fiber bound context storage, and hence required changes to [Context::storage](https://github.com/Nevay/opentelemetry-php/blob/5ff28f6dcc7a4002e67bd01291227283c5cf8b81/src/Context/Context.php#L44-L48). For backward compatibility reasons, we have to keep both initializations.
- To proactively spot such changes and ensure backward compatibility, the `test_opentelemetry_1` suite (currently: `@stable`) will be split into two: `@beta` and `1.0.*`.
- There are some white space changes in the PR 😬 Sorry, I didn't manage to remove them :-(
- I just changed the Fiber test back to .phpt; it was driving me crazy and I couldn't otherwise properly reset the context between tests (...I didn't have time to look further into it). This way, it's appropriately reset :-)

### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

**Comment by PROFeNoM** [comment]
I'm putting it back to draft. +7-9% performance regression for a simple check is simply too much. I'll try and optimize that a little bit.

**Comment by bwoebi** [comment]
@PROFeNoM Feels like the benchmarks are simply flaky.

File: src/DDTrace/OpenTelemetry/Context.php [hunk 1]
```diff
@@ -58,8 +61,13 @@ public static function setStorage(ContextStorageInterface $storage): void
      */
     public static function storage(): ContextStorageInterface
     {
-        /** @psalm-suppress RedundantPropertyInitializationCheck */
-        return self::$storage ??= new ContextStorage();
+        if (self::$storageClass === '') {
+            self::$storageClass = class_exists('\OpenTelemetry\Context\FiberBoundContextStorageExecutionAwareBC')
+                ? '\OpenTelemetry\Context\FiberBoundContextStorageExecutionAwareBC' // v1.1+
+                : '\OpenTelemetry\Context\ContextStorage';
```
Comment (bwoebi) [inline src/DDTrace/OpenTelemetry/Context.php:67]:
```suggestion
            self::$storageClass = class_exists('OpenTelemetry\Context\FiberBoundContextStorageExecutionAwareBC')
                ? 'OpenTelemetry\Context\FiberBoundContextStorageExecutionAwareBC' // v1.1+
                : 'OpenTelemetry\Context\ContextStorage';
```
Removing the leading slashes should optimize a little too.

**Review by bwoebi — APPROVED** [review]
Thanks, looks good :-)

**Pull request merged**